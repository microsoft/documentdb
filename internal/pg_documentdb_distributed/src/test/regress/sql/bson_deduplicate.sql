
SET search_path TO helio_api,helio_api_internal,helio_core;
SET citus.next_shard_id TO 4600000;
SET helio_api.next_collection_id TO 4600;
SET helio_api.next_collection_index_id TO 4600;

SELECT helio_api_internal.bson_deduplicate_fields(null);

SELECT helio_api_internal.bson_deduplicate_fields('{}') = '{}';

SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "a": 1}') = '{"a":1}';
SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "a": 1}') = '{"a":1}';

SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "b": [{"c": 1, "c": 2}, {"c": {"e": 1, "z": [], "e": 2}}], "a": null}') =
                                                  '{"a": null, "b" : [{"c": 2}, {"c": {"e": 2, "z": []}}]}';
SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "b": [{"c": 1, "c": [1, 1, "text", {"d": 1, "d": 2}]}, {"c": {"e": 1, "e": 2}}], "a": 2}') =
                                                  '{"a": 2, "b": [{"c": [1, 1, "text", {"d": 2}]}, {"c": {"e": 2}}]}';
SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "a.b": 2, "a.b.c": "text"}') = '{"a": 1, "a.b": 2, "a.b.c": "text"}';
