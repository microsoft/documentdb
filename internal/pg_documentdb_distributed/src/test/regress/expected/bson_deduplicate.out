SET search_path TO helio_api,helio_api_internal,helio_core;
SET citus.next_shard_id TO 4600000;
SET helio_api.next_collection_id TO 4600;
SET helio_api.next_collection_index_id TO 4600;
SELECT helio_api_internal.bson_deduplicate_fields(null);
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields(null);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{}') = '{}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{}') = '{...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "a": 1}') = '{"a":1}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "a": 1}') = '{"a":1}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{"a":1, "...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "b": [{"c": 1, "c": 2}, {"c": {"e": 1, "z": [], "e": 2}}], "a": null}') =
                                                  '{"a": null, "b" : [{"c": 2}, {"c": {"e": 2, "z": []}}]}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, ...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "b": [{"c": 1, "c": [1, 1, "text", {"d": 1, "d": 2}]}, {"c": {"e": 1, "e": 2}}], "a": 2}') =
                                                  '{"a": 2, "b": [{"c": [1, 1, "text", {"d": 2}]}, {"c": {"e": 2}}]}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, ...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, "a.b": 2, "a.b.c": "text"}') = '{"a": 1, "a.b": 2, "a.b.c": "text"}';
ERROR:  function helio_api_internal.bson_deduplicate_fields(unknown) does not exist
LINE 1: SELECT helio_api_internal.bson_deduplicate_fields('{"a": 1, ...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
